### PR TITLE
[4.0] Move to Dart SASS

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

Libsass and node-sass are deprecated, so this PR moves J4 over to Dart SASS. 

This will also fix not being able to run CSS related build scripts for Joomla 4 on NodeJS 15+

### Testing Instructions

Run `npm run build:css` from your terminal

### CODE REVIEW

**Comment by HLeithner** [comment]
please fix cs @C-Lodder thx

File: build/build-modules-js/stylesheets/scss-transform.es6.js [hunk 1]
```diff
@@ -4,54 +4,50 @@ const Fs = require('fs');
...
-      // Auto prefixing
-      // eslint-disable-next-line no-console
-      const cleaner = Postcss(
-        [
-          Autoprefixer(),
-        ],
-      );
+  let compiled;
+  try {
+    compiled = Sass.renderSync({ file: file });
```
Comment (dgrammatiko) [inline build/build-modules-js/stylesheets/scss-transform.es6.js]: Change it to `compiled = Sass.renderSync({ file });` to make ESLint happy

**Comment by brianteeman** [comment]
Is it correct that there is a new file in this PR `build/media_source/system/scss/_jalert.scss`

**Comment by SharkyKZ** [comment]
@brianteeman No, it should be removed.

**Comment by C-Lodder** [comment]
done

**Comment by ghazal** [comment]
I have tested this item :white_check_mark: successfully on 1620dc7b57ad7196dd463cb5282b8cdacf6392c3<br /><br />Tested OK providing that Composer dependency phpcodesniffer-composer-installer is updated as I use composer 2.
I don't PR this update as I can't test on a lesser version of Composer (i.e Composer 1.x). But it is ready if needed.

composer 2.0.4
npm 6.14.8
node v15.0.1

PHP 7.4.9
macOS Catalina<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31290">issues.joomla.org/tracker/joomla-cms/31290</a>.</sub>

**Comment by jwaisner** [comment]
@C-Lodder can you please correct the conflict in this PR?

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on f565079a3497888bf978211b29527cc4b119be7a<br /><br />No issues on NPM build. Checked a new install and everything functions as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31290">issues.joomla.org/tracker/joomla-cms/31290</a>.</sub>

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on f565079a3497888bf978211b29527cc4b119be7a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31290">issues.joomla.org/tracker/joomla-cms/31290</a>.</sub>

**Comment by SharkyKZ** [comment]
RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31290">issues.joomla.org/tracker/joomla-cms/31290</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**